src/component/FileManager/Dialogs/OpenWith.tsx

```diff
@@ -174,21 +179,26 @@ const OpenWith = () => {
               overflow: "auto",
             }}
           >
-            {((expanded ? Object.values(ViewersByID) : selectorState?.viewers) ?? emptyViewer).map((viewer) => (
-              <ListItem
-                disablePadding
-                key={viewer.id}
-                onDoubleClick={() => openWith(false, viewer)}
-                onClick={() => onViewerClick(viewer)}
-              >
-                <ListItemButton selected={viewer.id == selectedViewer?.id}>
-                  <ListItemAvatar sx={{ minWidth: "48px" }}>
-                    <ViewerIcon viewer={viewer} />
-                  </ListItemAvatar>
-                  <ListItemText primary={t(viewer.display_name)} />
-                </ListItemButton>
-              </ListItem>
-            ))}
+            {((expanded ? Object.values(ViewersByID) : selectorState?.viewers) ?? emptyViewer)
+              .filter((viewer) => {
+                const platform = viewer.platform || "all";
+                return platform === "all" || platform === (isMobile ? "mobile" : "pc");
```

Comment:
Those filtered viewers are not excluded in https://github.com/cloudreve/frontend/blob/master/src/component/FileManager/ContextMenu/OpenWithMenuItems.tsx

Maybe you can do the filtering here: https://github.com/cloudreve/frontend/blob/2c5611615392400ba6d90fd582042cfd7207ebe3/src/redux/siteConfigSlice.ts#L71

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia